### DISCUSSION TITLE
There is not an AStar algorithm that returns distance and path

### DISCUSSION THREAD

**Discussion by Xemorr** [networkx/networkx#6838] [Algorithms]
Hi, I cannot see an implementation of A* that returns distance and path, unlike dijkstra. I propose adding an implementation that does this, hopefully named "astar" to be in line with the naming conventions elsewhere.

**Comment by MridulS** [comment]
Thanks for opening the discussion @Xemorr 
There is already a [`astar_path`](https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.shortest_paths.astar.astar_path.html#networkx.algorithms.shortest_paths.astar.astar_path) and [`astar_path_length`](https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.shortest_paths.astar.astar_path_length.html#networkx.algorithms.shortest_paths.astar.astar_path_length) implementation. Is there anything missing in these implementations?

**Reply by Xemorr** [reply]
People normally use AStar over Dijkstras etc, for performance reasons. Distance is already being calculated in the implementation of astar_path, just not being returned to the user, this means the user needs to reiterate over the path and calculate it themselves, it would be better if there was an implementation of A* that returned both.
